tests/fixtures/joint.py

```diff
@@ -76,37 +76,29 @@ async def call_search_endpoint(
         return models.QueryResults(**response.json())
 
 
-async def joint_fixture_function(
-    mongodb_fixture: MongoDbFixture, kafka_fixture: KafkaFixture
+@pytest_asyncio.fixture
+async def joint_fixture(
+    mongodb: MongoDbFixture, kafka: KafkaFixture
 ) -> AsyncGenerator[JointFixture, None]:
-    """A fixture that embeds all other fixtures for API-level integration testing
-
-    **Do not call directly** Instead, use get_joint_fixture().
-    """
+    """A fixture that embeds all other fixtures for API-level integration testing."""
     # merge configs from different sources with the default one:
-    config = get_config(sources=[mongodb_fixture.config, kafka_fixture.config])
-
-    # create a DI container instance:translators
-    async with prepare_core(config=config) as query_handler:
-        async with (
-            prepare_rest_app(
-                config=config, query_handler_override=query_handler
-            ) as app,
-            prepare_event_subscriber(
-                config=config, query_handler_override=query_handler
-            ) as event_subscriber,
-        ):
-            async with AsyncTestClient(app=app) as rest_client:
-                yield JointFixture(
-                    config=config,
-                    query_handler=query_handler,
-                    event_subscriber=event_subscriber,
-                    kafka=kafka_fixture,
-                    mongodb=mongodb_fixture,
-                    rest_client=rest_client,
-                )
-
-
-def get_joint_fixture(scope: PytestScope = "function"):
-    """Produce a joint fixture with desired scope"""
-    return pytest_asyncio.fixture(joint_fixture_function, scope=scope)
+    config = get_config(sources=[mongodb.config, kafka.config])
+
+    async with (
+        prepare_core(config=config) as query_handler,
+        prepare_rest_app(config=config, query_handler_override=query_handler) as app,
+        prepare_event_subscriber(
+            config=config, query_handler_override=query_handler
+        ) as event_subscriber,
+        AsyncTestClient(app=app) as rest_client,
+    ):
+        joint_fixture = JointFixture(
+            config=config,
+            query_handler=query_handler,
+            event_subscriber=event_subscriber,
+            kafka=kafka,
+            mongodb=mongodb,
+            rest_client=rest_client,
+        )
+        await joint_fixture.load_test_data()
```

Comment:
Where was this handled before?

Reply:
Previously, there was a function-scope fixture that ran on every test case because we didn't have the built-in reset. (So the fixture would reset the mongodb state, then insert the test data). Since mongodb resets its state automatically now in the hexkit fixture, I moved the `load_test_data()` call to inside the joint fixture rather than keeping it in a separate fixture.